tidy3d/components/simulation.py

```diff
@@ -3009,6 +3043,18 @@ def num_cells(self) -> int:
 
         return np.prod(self.grid.num_cells, dtype=np.int64)
 
+    @cached_property
+    def freq_max(self) -> float:
+        """Maximum frequency in the ``Simulation``."""
+        return max(source.source_time.freq0 for source in self.sources)
+
+    @cached_property
+    def n_max(self) -> float:
+        """Maximum refractive index in the ``Simulation``."""
+        eps_max = max(abs(struct.medium.eps_model(self.freq_max)) for struct in self.structures)
```

Comment:
Is there any reason to use the maximum frequency instead of the central one? It is possible that the maximum index is found at lower frequencies, no?
Also, if you are looking for max(n), that should be max(Re{sqrt(ε)}), not Re{sqrt(argmax(|ε|))}.

Reply:
Hm, so this was basically refactored out of our previous `wvl_min_material`, which grabbed the minimum wavelength in the material. I guess I should probably do something like take the central frequency or average over frequencies.

> Also, if you are looking for max(n), that should be max(Re{sqrt(ε)}), not Re{sqrt(argmax(|ε|))}.

I wonder also if this means our previous function was wrong.

Reply:
ok, let's use central frequency and I'll just use `eps_complex_to_nk(eps_model(source_time.freq0))`